shell/platform/android/io/flutter/embedding/engine/systemchannels/TextInputChannel.java

```diff
@@ -112,6 +113,22 @@ public void onMethodCall(@NonNull MethodCall call, @NonNull MethodChannel.Result
               textInputMethodHandler.clearClient();
               result.success(null);
               break;
+            case "TextInput.sendAppPrivateCommand":
+              try {
+                final JSONObject arguments = (JSONObject) args;
+                final String action = arguments.getString("action");
+                final String data = arguments.getString("data");
+                Bundle bundle = null;
+                if (data != null && !data.isEmpty()) {
+                  bundle = new Bundle();
+                  bundle.putString("data", data);
```

Comment:
Do all private app commands just take a bundle with a single "data" key where the value is a string?

Reply:
For Crowdsource project, no "data" is needed to associate with the "action". But based on the Android equivalent API, "data" can be many types because it is a bundle type in Android. Do we need to support them all?

Reply:
I would think no, we don't need to support the other types, since this is such a niche API.  We can add them if/when someone has a use for them.

Though, I see you did add support for all datatypes in https://github.com/flutter/engine/pull/20188.